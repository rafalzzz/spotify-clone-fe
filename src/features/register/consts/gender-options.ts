import { TOptionType } from '@/types/custom-form-item-props';

export const GENDER_OPTIONS: TOptionType[] = [
  { label: 'Man', value: 1 },
  { label: 'Woman', value: 2 },
  { label: 'Non-binary person', value: 3 },
  { label: 'Other', value: 4 },
  { label: "I don't want to give", value: 5 },
];
