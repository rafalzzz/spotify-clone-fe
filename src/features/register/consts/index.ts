export { FORM_FIELDS } from './form-fields';
export { INITIAL_VALUES } from './initial-values';
export { GENDER_OPTIONS } from './gender-options';
export { MONTH_OPTIONS } from './month-options';
export { FORM_FIELD_PLACEHOLDERS } from './form-field-placeholders';
export { REGISTER_FORM_LABELS } from './form-field-labels';
