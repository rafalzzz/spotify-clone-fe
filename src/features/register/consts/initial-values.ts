import { ERegisterFormKeys } from '../types';

export const INITIAL_VALUES = {
  [ERegisterFormKeys.EMAIL]: '',
  [ERegisterFormKeys.PASSWORD]: '',
  [ERegisterFormKeys.NICKNAME]: '',
  [ERegisterFormKeys.DAY]: '',
  [ERegisterFormKeys.MONTH]: undefined,
  [ERegisterFormKeys.YEAR]: '',
  [ERegisterFormKeys.GENDER]: undefined,
  [ERegisterFormKeys.OFFERS]: false,
  [ERegisterFormKeys.SHARE_INFORMATION]: false,
  [ERegisterFormKeys.TERMS]: false,
};
