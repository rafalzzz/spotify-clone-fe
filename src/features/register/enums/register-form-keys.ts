export enum RegisterFormKeys {
  EMAIL = 'email',
  PASSWORD = 'password',
  NICKNAME = 'nickname',
  DAY = 'day',
  MONTH = 'month',
  YEAR = 'year',
  GENDER = 'gender',
  OFFERS = 'offers',
  SHARE_INFORMATION = 'shareInformation',
  TERMS = 'terms',
}

export enum RegisterFormAdditionalKeys {
  DATE_OF_BIRTH = 'dateOfBirth',
}
