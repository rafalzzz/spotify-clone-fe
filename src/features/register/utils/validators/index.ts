export { emailValidator } from './email-validator';
export { genderValidator } from './gender-validator';
export { usernameValidator } from './username-validator';
export { passwordValidator } from './password-validator';
export { dateOfBirthValidator } from './date-of-birth-validator';
export { termsValidator } from './terms-validator';
