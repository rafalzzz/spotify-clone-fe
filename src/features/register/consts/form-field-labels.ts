import { RegisterFormKeys } from '@/register/enums/register-form-keys';

import { NonStandardInputType } from '@/enums/input-type';

export const REGISTER_FORM_LABELS = {
  [RegisterFormKeys.EMAIL]: 'Your e-mail address',
  [RegisterFormKeys.PASSWORD]: 'Create a password',
  [RegisterFormKeys.NICKNAME]: 'How should we address you?',
  [NonStandardInputType.DATE_OF_BIRTH]: 'Enter your date of birth',
  [RegisterFormKeys.DAY]: 'Day',
  [RegisterFormKeys.MONTH]: 'Month',
  [RegisterFormKeys.YEAR]: 'Year',
  [RegisterFormKeys.GENDER]: 'Your gender?',
  [RegisterFormKeys.OFFERS]: 'I want to receive news and offers from Spotify',
  [RegisterFormKeys.SHARE_INFORMATION]:
    'Share my registration information with content providers on Spotify. This information may be used for marketing purposes. Note: According to our Privacy Policy, your data may be sent to a country outside the EEA.',
  [RegisterFormKeys.TERMS]: 'I accept the Spotify Terms of Use.',
};
