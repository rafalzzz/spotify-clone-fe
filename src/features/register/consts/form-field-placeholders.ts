import { RegisterFormKeys } from '@/register/enums/register-form-keys';

import { FORM_LABELS } from './form-field-labels';

export const FORM_FIELD_PLACEHOLDERS = {
  [RegisterFormKeys.EMAIL]: 'Give me your e-mail address',
  [RegisterFormKeys.PASSWORD]: FORM_LABELS[RegisterFormKeys.PASSWORD],
  [RegisterFormKeys.NICKNAME]: 'Enter a username',
  [RegisterFormKeys.DAY]: 'DD',
  [RegisterFormKeys.MONTH]: 'Month',
  [RegisterFormKeys.YEAR]: 'RRRR',
};
