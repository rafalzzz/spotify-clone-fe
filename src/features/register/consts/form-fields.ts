import { RegisterFormKeys } from '@/register/enums/register-form-keys';
import {
  emailValidator,
  usernameValidator,
  passwordValidator,
  genderValidator,
  termsValidator,
} from '@/register/utils/validators';

import { InputType, NonStandardInputType } from '@/enums/input-type';

import { CustomFormItemProps } from '@/types/custom-form-item-props';
import { NonStandardItemProps } from '@/types/non-standard-form-item-props';

import { FORM_LABELS } from './form-field-labels';
import { FORM_FIELD_PLACEHOLDERS } from './form-field-placeholders';
import { GENDER_OPTIONS } from './gender-options';

export const FORM_FIELDS: (CustomFormItemProps<RegisterFormKeys> | NonStandardItemProps)[] = [
  {
    type: InputType.TEXT,
    key: RegisterFormKeys.EMAIL,
    name: RegisterFormKeys.EMAIL,
    label: FORM_LABELS[RegisterFormKeys.EMAIL],
    rules: [{ validator: emailValidator }],
    inputProps: {
      placeholder: FORM_FIELD_PLACEHOLDERS[RegisterFormKeys.EMAIL],
    },
  },
  {
    type: InputType.PASSWORD,
    key: RegisterFormKeys.PASSWORD,
    name: RegisterFormKeys.PASSWORD,
    label: FORM_LABELS[RegisterFormKeys.PASSWORD],
    rules: [{ validator: passwordValidator }],
    inputProps: {
      placeholder: FORM_LABELS[RegisterFormKeys.PASSWORD],
    },
  },
  {
    type: InputType.TEXT,
    key: RegisterFormKeys.NICKNAME,
    name: RegisterFormKeys.NICKNAME,
    label: FORM_LABELS[RegisterFormKeys.NICKNAME],
    rules: [{ validator: usernameValidator }],
    inputProps: {
      placeholder: FORM_FIELD_PLACEHOLDERS[RegisterFormKeys.NICKNAME],
    },
  },
  {
    type: NonStandardInputType.DATE_OF_BIRTH,
    key: NonStandardInputType.DATE_OF_BIRTH,
    name: NonStandardInputType.DATE_OF_BIRTH,
    label: FORM_LABELS[NonStandardInputType.DATE_OF_BIRTH],
  },
  {
    type: InputType.RADIO,
    key: RegisterFormKeys.GENDER,
    name: RegisterFormKeys.GENDER,
    label: FORM_LABELS[RegisterFormKeys.GENDER],
    rules: [{ validator: genderValidator }],
    radioProps: {
      options: GENDER_OPTIONS,
    },
  },
  {
    type: InputType.CHECKBOX,
    key: RegisterFormKeys.OFFERS,
    name: RegisterFormKeys.OFFERS,
    checkboxProps: {
      label: FORM_LABELS[RegisterFormKeys.OFFERS],
    },
  },
  {
    type: InputType.CHECKBOX,
    key: RegisterFormKeys.SHARE_INFORMATION,
    name: RegisterFormKeys.SHARE_INFORMATION,
    checkboxProps: {
      label: FORM_LABELS[RegisterFormKeys.SHARE_INFORMATION],
    },
  },
  {
    type: InputType.CHECKBOX,
    key: RegisterFormKeys.TERMS,
    name: RegisterFormKeys.TERMS,
    rules: [{ validator: termsValidator }],
    checkboxProps: {
      label: FORM_LABELS[RegisterFormKeys.TERMS],
    },
  },
];
