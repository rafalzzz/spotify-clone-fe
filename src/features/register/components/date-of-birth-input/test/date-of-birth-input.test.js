import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { Form } from 'antd';

import { FORM_FIELD_PLACEHOLDERS, REGISTER_FORM_LABELS } from '@/register/consts';
import { RegisterFormKeys } from '@/register/enums/register-form-keys';
import { dateOfBirthValidator } from '@/register/utils/validators';

import { InputType, NonStandardInputType } from '@/enums/input-type';

import { DateOfBirthInput } from '../index.tsx';

jest.mock('@/register/utils/validators', () => ({
  dateOfBirthValidator: jest.fn().mockImplementation((getFieldValue) => () => Promise.resolve()),
}));

const renderDateOfBirthInput = () =>
  render(
    <Form>
      <DateOfBirthInput
        name={NonStandardInputType.DATE_OF_BIRTH}
        label={REGISTER_FORM_LABELS[NonStandardInputType.DATE_OF_BIRTH]}
      />
    </Form>,
  );

describe('DateOfBirthInput', () => {
  it('render component without error', () => {
    const screen = renderDateOfBirthInput();
    expect(screen).toMatchSnapshot();
  });

  describe('should call validateFieldsMock when input value change', () => {
    const formFieldsWithValidators = [
      {
        type: InputType.TEXT,
        key: RegisterFormKeys.DAY,
        placeholder: FORM_FIELD_PLACEHOLDERS[RegisterFormKeys.DAY],
        mockedValue: '1',
      },
      {
        type: InputType.SELECT,
        key: RegisterFormKeys.MONTH,
        mockedValue: 'January',
      },
      {
        type: InputType.TEXT,
        key: RegisterFormKeys.YEAR,
        placeholder: FORM_FIELD_PLACEHOLDERS[RegisterFormKeys.YEAR],
        mockedValue: '2000',
      },
    ];

    formFieldsWithValidators.forEach(({ type, key, validator, ...restProps }) => {
      it(key, async () => {
        const { getByRole, getByLabelText, getByPlaceholderText } = renderDateOfBirthInput();

        if (type === InputType.SELECT) {
          const { mockedValue } = restProps;

          const select = getByRole('combobox');
          expect(select).toBeInTheDocument();

          await userEvent.click(select);

          const monthOptions = await getByRole('listbox');

          const option = getByRole('option', { name: mockedValue });
          await userEvent.click(option);

          expect(dateOfBirthValidator).toHaveBeenCalled();
          return;
        }

        const { mockedValue, placeholder } = restProps;
        const input = getByPlaceholderText(placeholder);
        expect(input).toBeInTheDocument();

        await userEvent.type(input, mockedValue);

        expect(dateOfBirthValidator).toHaveBeenCalled();
      });
    });
  });
});
