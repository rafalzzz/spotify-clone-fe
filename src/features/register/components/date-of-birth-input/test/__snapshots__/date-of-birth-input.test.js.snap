// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateOfBirthInput render component without error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="ant-form ant-form-horizontal css-dev-only-do-not-override-12upa3x"
      >
        <div
          class="ant-form-item date-of-birth css-dev-only-do-not-override-12upa3x"
        >
          <div
            class="ant-row ant-form-item-row css-dev-only-do-not-override-12upa3x"
          >
            <div
              class="ant-col ant-form-item-label css-dev-only-do-not-override-12upa3x"
            >
              <label
                class=""
                for="dateOfBirth"
                title=""
              >
                <span
                  class="date-of-birth__label"
                >
                  Enter your date of birth
                </span>
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control css-dev-only-do-not-override-12upa3x"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-row css-dev-only-do-not-override-12upa3x"
                    id="dateOfBirth"
                  >
                    <div
                      class="ant-col ant-col-4 css-dev-only-do-not-override-12upa3x"
                    >
                      <div
                        class="ant-form-item date-of-birth__input css-dev-only-do-not-override-12upa3x"
                      >
                        <div
                          class="ant-row ant-form-item-row css-dev-only-do-not-override-12upa3x"
                        >
                          <div
                            class="ant-col ant-form-item-label css-dev-only-do-not-override-12upa3x"
                          >
                            <label
                              class=""
                              for="day"
                              title=""
                            >
                              <span
                                class="date-of-birth__label"
                              >
                                Day
                              </span>
                            </label>
                          </div>
                          <div
                            class="ant-col ant-form-item-control css-dev-only-do-not-override-12upa3x"
                          >
                            <div
                              class="ant-form-item-control-input"
                            >
                              <div
                                class="ant-form-item-control-input-content"
                              >
                                <input
                                  class="ant-input css-dev-only-do-not-override-12upa3x"
                                  id="day"
                                  maxlength="2"
                                  placeholder="DD"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-col ant-col-14 ant-col-offset-1 css-dev-only-do-not-override-12upa3x"
                    >
                      <div
                        class="ant-form-item date-of-birth__input css-dev-only-do-not-override-12upa3x"
                      >
                        <div
                          class="ant-row ant-form-item-row css-dev-only-do-not-override-12upa3x"
                        >
                          <div
                            class="ant-col ant-form-item-label css-dev-only-do-not-override-12upa3x"
                          >
                            <label
                              class=""
                              for="Month"
                              title=""
                            >
                              <span
                                class="date-of-birth__label"
                              >
                                Month
                              </span>
                            </label>
                          </div>
                          <div
                            class="ant-col ant-form-item-control css-dev-only-do-not-override-12upa3x"
                          >
                            <div
                              class="ant-form-item-control-input"
                            >
                              <div
                                class="ant-form-item-control-input-content"
                              >
                                <div
                                  class="ant-select ant-select-in-form-item css-dev-only-do-not-override-12upa3x ant-select-single ant-select-show-arrow"
                                >
                                  <div
                                    class="ant-select-selector"
                                  >
                                    <span
                                      class="ant-select-selection-search"
                                    >
                                      <input
                                        aria-activedescendant="Month_list_0"
                                        aria-autocomplete="list"
                                        aria-controls="Month_list"
                                        aria-expanded="false"
                                        aria-haspopup="listbox"
                                        aria-owns="Month_list"
                                        autocomplete="off"
                                        class="ant-select-selection-search-input"
                                        id="Month"
                                        readonly=""
                                        role="combobox"
                                        style="opacity: 0;"
                                        type="search"
                                        unselectable="on"
                                        value=""
                                      />
                                    </span>
                                    <span
                                      class="ant-select-selection-placeholder"
                                    >
                                      Month
                                    </span>
                                  </div>
                                  <span
                                    aria-hidden="true"
                                    class="ant-select-arrow"
                                    style="user-select: none;"
                                    unselectable="on"
                                  >
                                    <span
                                      aria-label="down"
                                      class="anticon anticon-down ant-select-suffix"
                                      role="img"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="down"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="64 64 896 896"
                                        width="1em"
                                      >
                                        <path
                                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                        />
                                      </svg>
                                    </span>
                                  </span>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-col ant-col-4 ant-col-offset-1 css-dev-only-do-not-override-12upa3x"
                    >
                      <div
                        class="ant-form-item date-of-birth__input css-dev-only-do-not-override-12upa3x"
                      >
                        <div
                          class="ant-row ant-form-item-row css-dev-only-do-not-override-12upa3x"
                        >
                          <div
                            class="ant-col ant-form-item-label css-dev-only-do-not-override-12upa3x"
                          >
                            <label
                              class=""
                              for="year"
                              title=""
                            >
                              <span
                                class="date-of-birth__label"
                              >
                                Year
                              </span>
                            </label>
                          </div>
                          <div
                            class="ant-col ant-form-item-control css-dev-only-do-not-override-12upa3x"
                          >
                            <div
                              class="ant-form-item-control-input"
                            >
                              <div
                                class="ant-form-item-control-input-content"
                              >
                                <input
                                  class="ant-input css-dev-only-do-not-override-12upa3x"
                                  id="year"
                                  maxlength="4"
                                  placeholder="RRRR"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="ant-form ant-form-horizontal css-dev-only-do-not-override-12upa3x"
    >
      <div
        class="ant-form-item date-of-birth css-dev-only-do-not-override-12upa3x"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-12upa3x"
        >
          <div
            class="ant-col ant-form-item-label css-dev-only-do-not-override-12upa3x"
          >
            <label
              class=""
              for="dateOfBirth"
              title=""
            >
              <span
                class="date-of-birth__label"
              >
                Enter your date of birth
              </span>
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-12upa3x"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  class="ant-row css-dev-only-do-not-override-12upa3x"
                  id="dateOfBirth"
                >
                  <div
                    class="ant-col ant-col-4 css-dev-only-do-not-override-12upa3x"
                  >
                    <div
                      class="ant-form-item date-of-birth__input css-dev-only-do-not-override-12upa3x"
                    >
                      <div
                        class="ant-row ant-form-item-row css-dev-only-do-not-override-12upa3x"
                      >
                        <div
                          class="ant-col ant-form-item-label css-dev-only-do-not-override-12upa3x"
                        >
                          <label
                            class=""
                            for="day"
                            title=""
                          >
                            <span
                              class="date-of-birth__label"
                            >
                              Day
                            </span>
                          </label>
                        </div>
                        <div
                          class="ant-col ant-form-item-control css-dev-only-do-not-override-12upa3x"
                        >
                          <div
                            class="ant-form-item-control-input"
                          >
                            <div
                              class="ant-form-item-control-input-content"
                            >
                              <input
                                class="ant-input css-dev-only-do-not-override-12upa3x"
                                id="day"
                                maxlength="2"
                                placeholder="DD"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-col ant-col-14 ant-col-offset-1 css-dev-only-do-not-override-12upa3x"
                  >
                    <div
                      class="ant-form-item date-of-birth__input css-dev-only-do-not-override-12upa3x"
                    >
                      <div
                        class="ant-row ant-form-item-row css-dev-only-do-not-override-12upa3x"
                      >
                        <div
                          class="ant-col ant-form-item-label css-dev-only-do-not-override-12upa3x"
                        >
                          <label
                            class=""
                            for="Month"
                            title=""
                          >
                            <span
                              class="date-of-birth__label"
                            >
                              Month
                            </span>
                          </label>
                        </div>
                        <div
                          class="ant-col ant-form-item-control css-dev-only-do-not-override-12upa3x"
                        >
                          <div
                            class="ant-form-item-control-input"
                          >
                            <div
                              class="ant-form-item-control-input-content"
                            >
                              <div
                                class="ant-select ant-select-in-form-item css-dev-only-do-not-override-12upa3x ant-select-single ant-select-show-arrow"
                              >
                                <div
                                  class="ant-select-selector"
                                >
                                  <span
                                    class="ant-select-selection-search"
                                  >
                                    <input
                                      aria-activedescendant="Month_list_0"
                                      aria-autocomplete="list"
                                      aria-controls="Month_list"
                                      aria-expanded="false"
                                      aria-haspopup="listbox"
                                      aria-owns="Month_list"
                                      autocomplete="off"
                                      class="ant-select-selection-search-input"
                                      id="Month"
                                      readonly=""
                                      role="combobox"
                                      style="opacity: 0;"
                                      type="search"
                                      unselectable="on"
                                      value=""
                                    />
                                  </span>
                                  <span
                                    class="ant-select-selection-placeholder"
                                  >
                                    Month
                                  </span>
                                </div>
                                <span
                                  aria-hidden="true"
                                  class="ant-select-arrow"
                                  style="user-select: none;"
                                  unselectable="on"
                                >
                                  <span
                                    aria-label="down"
                                    class="anticon anticon-down ant-select-suffix"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="down"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-col ant-col-4 ant-col-offset-1 css-dev-only-do-not-override-12upa3x"
                  >
                    <div
                      class="ant-form-item date-of-birth__input css-dev-only-do-not-override-12upa3x"
                    >
                      <div
                        class="ant-row ant-form-item-row css-dev-only-do-not-override-12upa3x"
                      >
                        <div
                          class="ant-col ant-form-item-label css-dev-only-do-not-override-12upa3x"
                        >
                          <label
                            class=""
                            for="year"
                            title=""
                          >
                            <span
                              class="date-of-birth__label"
                            >
                              Year
                            </span>
                          </label>
                        </div>
                        <div
                          class="ant-col ant-form-item-control css-dev-only-do-not-override-12upa3x"
                        >
                          <div
                            class="ant-form-item-control-input"
                          >
                            <div
                              class="ant-form-item-control-input-content"
                            >
                              <input
                                class="ant-input css-dev-only-do-not-override-12upa3x"
                                id="year"
                                maxlength="4"
                                placeholder="RRRR"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
